### PR TITLE
Add (optional) blockedCountries list for notifications

### PR DESCRIPTION
**Pull request by anton-clabs**
### Description

Currently there is no easy way to exclude certain notifications from a set of countries: the workaround is adding every OTHER country to the countries list for that notification, which is clearly cumbersome and inefficient when we only want to exclude a small number of countries (e.g., creating a notification to be shown everywhere except the Philippines). The present PR adds the option to include a blocked countries list in a notification interface, and checks a user's country to ensure it is not on a notification's blocked list before sending the notification.

Also adds two test cases: a notification with PH blocked and a notification with FR blocked (https://github.com/celo-org/wallet/blob/1b13812dd0d4ab0b99fa7658ecdd931009c4d234/packages/mobile/src/home/selectors.test.ts).

### Other changes

N/A

### Tested

Tested with the new blocked-list-specific cases in command-line using jest.

### Related issues

- Fixes https://github.com/celo-org/celo-monorepo/issues/7018

### Backwards compatibility

Inherited from compatibility for notifications v2

### CODE REVIEW

File: packages/mobile/src/home/reducers.ts [hunk 1]
```diff
@@ -17,6 +17,7 @@ export interface Notification {
   minVersion?: string
   maxVersion?: string
   countries?: string[]
+  blacklist?: string[]
```
Comment (jeanregisser) [inline packages/mobile/src/home/reducers.ts:20]: We should avoid `blacklist`, see https://tools.ietf.org/id/draft-knodel-terminology-00.html#rfc.section.1.2
How about `blockedCountries`?

**Review by jeanregisser — CHANGES_REQUESTED** [review]
Thanks!
Can you also update the test https://github.com/celo-org/wallet/blob/1b13812dd0d4ab0b99fa7658ecdd931009c4d234/packages/mobile/src/home/selectors.test.ts?

File: packages/mobile/src/home/reducers.ts (hunk 1 above)
Comment (anton-clabs) [reply packages/mobile/src/home/reducers.ts:20]: Thanks for pointing that out, I have often wondered about the usage of that term. Will make that change!

**Review by gnardini — APPROVED** [review]
Tested this using and changing values on Firebase, the notification shows up/disappears as expected.

Great job! 👏

**Review by jeanregisser — APPROVED** [review]
Awesome work! 👍

**Pull request merged**